### ISSUE TITLE
[bug] Directories with spaces show weird behaviour

### ISSUE THREAD

**Issue by kreczko** [rootpy/rootpy#663]
Today I encountered weird behaviour in the File/Directory classes.
It is best demonstrated by this example:
```python
from rootpy.io import File

f = File('test.root', 'recreate')
f.mkdir('TTbar_plus_X_analysis/EPlusJets/Ref selection', recurse=True)
f.mkdir('Ref selection')

# this one works
print 'Ref selection', f.__contains__('Ref selection')
# this one doesn't
print 'TTbar_plus_X_analysis/EPlusJets/Ref selection', f.__contains__('TTbar_plus_X_analysis/EPlusJets/Ref selection')

print list(f.walk())
```
In general directories/paths with spaces are fine. However, once you add subfolders, this is no longer the case. In the above example, rootpy is happy with ```Ref selection``` but not with ```TTbar_plus_X_analysis/EPlusJets/Ref selection```. 
Similar behaviour can be seen in ```roosh``` where a ```cd``` into the directory works, but an ```ls``` fails.

I am going to have a second look at the code tomorrow, but if any of you already know where the buggy code might reside, please let me know.

Cheers,
Luke

**Comment by ndawe** [comment]
The space isn't the problem since this also fails:

```python
from rootpy.io import File

f = File('test.root', 'recreate')
f.mkdir('a/b/c', recurse=True)
print 'a/b/c' in f
```

**Comment by ndawe** [comment]
Thanks for catching this bug. I'm surprised this survived so long. Fixed in #664
